polyfill/lib/calendar.mjs

```diff
@@ -185,6 +190,9 @@ impl['iso8601'] = {
   fields(fields) {
     return fields;
   },
+  mergeFields(fields, extraFields) {
+    return ObjectAssign(fields, extraFields);
```

Comment:
Will it be spec to always modify the first bag? To only call mergeFields on a potentially user-derived but not user-created bag? (i.e. how it's called on ToTemporal____Fields throughout the rest of this PR)